### PR TITLE
Expand the view panel in stock operation

### PR DESCRIPTION
**Pull request by gabriel090**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [ ] My work includes tests or is validated by existing tests.
Expand the view panel in the stock management system
Modified UI

https://github.com/user-attachments/assets/75ef40a2-37cb-43b8-a092-37a0ac9de133

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks, @gabriel090. Please link the ticket for this task. 
We may need to refine the interface a bit. The result panel should not push down other parts UI elements. Also, I feel the additional search button is redundant. THe on select event should be sufficient to set the selected item

**Comment by gabriel090** [comment]
Modified UI
https://github.com/user-attachments/assets/70f69805-d1c0-4baf-ac4c-231041b88eb7

**Comment by makombe** [comment]
@gabriel090 please address the lint issue failing for the build. Thanks

**Comment by makombe** [comment]
Also, updated the video indicating completion of the operation you are doing.

**Comment by ojwanganto** [comment]
This will really improve UX for products with long names. Kindly fix the build issues and share a video on the complete process

**Comment by makombe** [comment]
@gabriel090, it is a bad practice to close a pr with reviews already. We are likely to loose track of such. You just need to build on this one, unless there are compelling reason to do so. Thanks

**Comment by gabriel090** [comment]
> @gabriel090, it is a bad practice to close a pr with reviews already. We are likely to loose track of such. You just need to build on this one, unless there are compelling reasons to do so. Thanks

I have reopened the PR

**Comment by ojwanganto** [comment]
@hadijahkyampeire @jabahum @makombe @donaldkibet @slubwama - bringing this to your attention and action.

**Comment by donaldkibet** [comment]
Please fix the conflicts

File: src/core/api/types/stockOperation/StockOperationType.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export enum OperationType {
 }
 
 export function operationFromString(str: string): OperationType | undefined {
-  str = str.toLowerCase();
+  str = str?.toLowerCase();
```
Comment (ojwanganto) [inline src/core/api/types/stockOperation/StockOperationType.ts:39]: we need a proper name for the **str** variable. It is OK if this is alright

File: src/stock-operations/add-stock-operation/stock-items-addition-row.component.tsx [hunk 2]
```diff
@@ -130,7 +130,9 @@ const StockItemsAdditionRow: React.FC<StockItemsAdditionRowProps> = ({
       }
     }
   };
-
+  const isStockItem = (obj: any): obj is StockItemDTO => {
+    return typeof obj === "object" && obj !== null && "drugName" in obj;
```
Comment (ojwanganto) [inline src/stock-operations/add-stock-operation/stock-items-addition-row.component.tsx:134]: stock item should also include non-drugs. I hope this check will still hold

**Review by jabahum — COMMENTED** [review]
@gabriel090 lets fix these conflicts

File: src/stock-operations/add-stock-operation/stock-items-addition.component.tsx [hunk 3]
```diff
@@ -162,112 +182,120 @@ const StockItemsAddition: React.FC<StockItemsAdditionProps> = ({
...
-                          <Button
-                            name="save"
-                            type="button"
-                            className="submitButton"
-                            onClick={() => handleSubmit(handleSave)()}
-                            kind="primary"
-                            renderIcon={ArrowRight}
+                          <div
+                            style={{
+                              width: "3% !important",
```
Comment (jabahum) [inline src/stock-operations/add-stock-operation/stock-items-addition.component.tsx:240]: can we move this to a .sccs file

File: src/stock-operations/add-stock-operation/stock-items-addition.component.tsx [hunk 4]
```diff
@@ -162,112 +182,120 @@ const StockItemsAddition: React.FC<StockItemsAdditionProps> = ({
...
-                    itemUoM={itemUoM}
-                    fields={fields}
-                  />{" "}
-                </TableBody>
-              </Table>
-            </TableContainer>
-          )}
-        ></DataTable>
+                            {/* <Button
+                              renderIcon={Add}
```
Comment (jabahum) [inline src/stock-operations/add-stock-operation/stock-items-addition.component.tsx:247]: lets remove unnecessary code

File: src/stock-operations/add-stock-operation/stock-items-addition.component.tsx (hunk 3 above)
Comment (gabriel090) [reply src/stock-operations/add-stock-operation/stock-items-addition.component.tsx]: worked on it

File: src/stock-operations/add-stock-operation/stock-items-addition.component.tsx (hunk 4 above)
Comment (gabriel090) [reply src/stock-operations/add-stock-operation/stock-items-addition.component.tsx]: cleaned

**Comment by makombe** [comment]
@jabahum can you help with reviewing this again. Thanks

**Comment by jabahum** [comment]
@makombe on it

**Review by jabahum — COMMENTED** [review]
image
Should we have this editable even when the status of the operation is complete? cc @makombe , @donaldkibet .
it also seems like the searchable items are limited here. see video below
Before

https://github.com/user-attachments/assets/c38599cf-d587-467a-aa7c-ab290f1e3c4d

After 

https://github.com/user-attachments/assets/5749b9b6-cc01-4952-9852-b98d74fa7872

**Comment by ojwanganto** [comment]
@jabahum @makombe @gabriel090 - could we close on this today?

**Review by jabahum — APPROVED** [review]

**Pull request merged**